mne/beamformer/_lcmv.py

```diff
@@ -133,70 +133,66 @@ def _apply_lcmv(data, info, tmin, forward, reg, noise_cov=None,
-        if weight_norm is not None and pick_ori == 'max-power':
-            # finding optimal orientation for NAI and unit-noise gain
-            # based on [2]_, Eq. 4.47
-            tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
-            eig_vals, eig_vecs = linalg.eig(np.dot(linalg.pinv(tmp),
-                                                   np.dot(Wk, Gk)))
-            idx_max = eig_vals.argmax()
-            max_ori = eig_vecs[:, idx_max]
-            Wk[:] = np.dot(max_ori, Wk)
-            Gk = np.dot(Gk, max_ori)
-
-            # compute spatial filter for NAI
-            tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
-            denom = np.sqrt(tmp)
-            Wk /= denom
-            if weight_norm == 'nai':
-                Wk /= np.sqrt(noise)
-
-            is_free_ori = False
-
-        # NAI without orientation selection
-        elif weight_norm == 'nai' and pick_ori != 'max-power':
-            raise ValueError('Weight normalization with neural activity index '
-                             'is not implemented yet with free or fixed '
-                             'orientation.')
+        if pick_ori == 'max-power':
+            # weight normalization and orientation selection:
+            if weight_norm is not None and pick_ori == 'max-power':
+                # finding optimal orientation for NAI and unit-noise gain
+                # based on [2]_, Eq. 4.47
+                tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
+                eig_vals, eig_vecs = linalg.eig(np.dot(linalg.pinv(tmp),
+                                                       np.dot(Wk, Gk)))
+                idx_max = eig_vals.argmax()
+                max_ori = eig_vecs[:, idx_max]
+                Wk[:] = np.dot(max_ori, Wk)
+                Gk = np.dot(Gk, max_ori)
+
+                # compute spatial filter for NAI
+                tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
+                denom = np.sqrt(tmp)
+                Wk /= denom
+                if weight_norm == 'nai':
+                    Wk /= np.sqrt(noise)
+
+                is_free_ori = False
+
+            # no weight-normalization and max-power is not implemented yet:
+            else:
+                raise ValueError('The max-power orientation selection is not '
+                                 'yet implemented with weight_norm set to'
+                                 'None.')
 
-        # all other combinations are handled here:
+        # free orientation and normal orientation
         else:
-            if pick_ori == 'max-power':
-                raise ValueError('The unit-gain constraint beamformer is not '
-                                 'yet implemented with max-power orientation '
-                                 'selection.')
-
+            # compute the filters:
             if is_free_ori:
                 # Free source orientation
                 Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
             else:
                 # Fixed source orientation
```

Comment:
@britta-wstnr @agramfort Please note it is SUPER important to distinguish "unit noise gain" and "unit gain". "Unit noise gain" is what I believe you are discussing here, i.e., the weight normalized beamformer and the similar NAI, while "unit gain" is basically the original "standard" LCMV beamformer.